bin/hdfcoinc/pycbc_stat_dtphase2

```diff
@@ -0,0 +1,149 @@
+                wkey = list(nkey)
+                wkey[index] += b
+                wkey = tuple(wkey)
+                
+                if wkey in data:
+                    weight += data[wkey] * w
+                
+            nweights[tuple(nkey)] = weight
+            
+    return nweights
+    
+d = {ifo: pycbc.detector.Detector(ifo) for ifo in args.ifos}
+
+pycbc.init_logging(args.verbose)
+
+numpy.random.seed(args.seed)
+size = 1000000
+chunks = int(args.sample_size / size) + 1
+
+l = 0
+nsamples = 0
+weights = {}
+for k in range(chunks):
+    nsamples += size
+    logging.info('generating %s samples' % size)
+
+    # Choose random sky location and polarizations from
+    # an isotropic population
+    ra = uniform(0, 2 * numpy.pi, size=size)
+    dec = numpy.arccos(uniform(-1., 1., size=size)) - numpy.pi/2
+    inc = numpy.arccos(uniform(-1., 1., size=size))
+    pol = uniform(0, 2 * numpy.pi, size=size)
+    ip = numpy.cos(inc)
+    ic = 0.5 * (1.0 + ip * ip)
+
+    twidth = .001  # approximate timing error at lower SNRs
+    pwidth = .25   # approximate phase error at lower SNRs
+    serr = 1.4   #
+    sref = 5.0     # Reference SNR for bin smoothing
+    swidth = serr / sref
+
+    # calculate the toa, poa, and amplitude of each sample
+    data = {}
+    for r, ifo in enumerate(args.ifos):
+        data[ifo] = {}
+        fp, fc = d[ifo].antenna_pattern(ra, dec, pol, 0)
+        sp, sc = fp * ip, fc * ic
+        data[ifo]['s'] = ((sp)**2.0 + (sc)**2.0) ** 0.5 * args.relative_sensitivities[r]
+        data[ifo]['t'] = d[ifo].time_delay_from_earth_center(ra, dec, 0)
+        data[ifo]['p'] = numpy.arctan2(sc, sp)
+
+    # Bin the data
+    bind = []
+    keep = None
+    for i in range(len(args.ifos) -1):
+        ifo0 = args.ifos[0 + i]
+        ifo1 = args.ifos[1 + i]
+        dt = (data[ifo0]['t'] - data[ifo1]['t'])
+        dp = (data[ifo0]['p'] - data[ifo1]['p']) % (numpy.pi * 2.0)
+        sr = (data[ifo1]['s'] / data[args.ifos[0]]['s'])
+        dtbin = (dt / twidth).astype(numpy.int)
+        dpbin = (dp / pwidth).astype(numpy.int)
+        srbin = (sr / swidth).astype(numpy.int)
+        
+        srbmax = int((args.snr_ratio / swidth))
+        srbmin = int((1.0 / args.snr_ratio) / swidth)
+        
+        # We'll only store a limited range of ratios
+        if keep is None:
+            keep = (srbin < srbmax) & srbin > (srbmin)
+        else:
+            keep = keep & (srbin < srbmax) & srbin > (srbmin)
+        bind += [dtbin, dpbin, srbin]
+ 
+    # Calculate and sum the weights for each bin
+    # use first ifo as reference for weights
+    bind = [a[keep] for a in bind]
+    
+    w = data[args.ifos[0]]['s'][keep] ** 3            
+    for i, key in enumerate(zip(*bind)):
+        if key not in weights:
+            weights[key] = 0
+        weights[key] += w[i]
+        
+    ol = l
+    l = len(weights.values())
+    logging.info('%s, %s, %s, %s', l, l - ol, (l - ol) / float(size), l / float(nsamples))
```

Comment:
Could add more commenting/clarification of logging and more obvious naming of variables to make this clearer